infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -297,6 +297,7 @@ public void reset() {
     @Override
     public void reboot() {
```

Comment:
Just pushed some commits. Turning reboot into synchronized doesn't help, because I think that only restricts access to the reboot method, whereas we want to restrict access across methods. I implemented a new lock instead. As a first pass, I locked large portions of the code, but I might do a closer pass to lock less code.